src/extension/wps/web-wps/src/main/java/org/geoserver/wps/web/ProcessStatusProvider.java

```diff
@@ -65,4 +96,139 @@ public Object getPropertyValue(ExecutionStatus item) {
+                               prop.equals(USER)) {
+                                likes.add(FF.like(FF.property(prop.getName()), "*"+word+"*"));
+                            }
+                            //TODO: support other properties
+                                
+                        }
+                    }
+                }
+                ret = FF.or(likes);
+            }
+        }
+        return ret;
+    }
+
+    @Override
+    protected List<ExecutionStatus> getFilteredItems() {
+        ProcessStatusTracker tracker = GeoServerApplication.get().getBeanOfType(
+                ProcessStatusTracker.class);
+        ProcessStatusStore store = tracker.getStore();
+        Query query = new Query("status", getFilter());
+        if(count>0) {
+            query.setStartIndex(first);
+            query.setMaxFeatures(count);
+        }
+        SortParam sort = getSort();
+        if(sort!=null) {
+            SortByImpl[] sortBys = new SortByImpl[1];
+            final Property<?> property = getProperty(sort);
+            if(property.isSearchable()) {//we really need another flag
+
+                //hard to believe there isn't an implementing class!
+                sortBys[0] = new SortByImpl(new PropertyName() {
+
+                    @Override
+                    public <T> T evaluate(Object object, Class<T> context) {
+                        // TODO Auto-generated method stub
+                        return null;
+                    }
+
+                    @Override
+                    public Object evaluate(Object object) {
+                        // TODO Auto-generated method stub
+                        return null;
+                    }
+
+                    @Override
+                    public Object accept(ExpressionVisitor visitor, Object extraData) {
+                        // TODO Auto-generated method stub
+                        return null;
+                    }
+
+                    @Override
+                    public String getPropertyName() {
+                        // TODO Auto-generated method stub
+                        return property.getName();
+                    }
+
+                    @Override
+                    public NamespaceSupport getNamespaceContext() {
+                        // TODO Auto-generated method stub
+                        return null;
+                    }
+                }, SortOrder.ASCENDING);
+
+                if(sort.isAscending()) {
+                    sortBys[0].setSortOrder(SortOrder.ASCENDING);
+                }else {
+                    sortBys[0].setSortOrder(SortOrder.DESCENDING);
+                }
+                query.setSortBy(sortBys);
+
+            }
+        }
+        LOGGER.info("built query "+query+" to filter statuses");
+        return store.list(query);
+    }
+
+    @Override
+    public Iterator<ExecutionStatus> iterator(int first, int count) {
+        this.first = first;
```

Comment:
This is not thread safe... but I'm wondering if we should care, wicket pages should be locked in the http session.